R/doeResponseSurfaceMethodology.R

```diff
@@ -21,34 +21,57 @@ doeResponseSurfaceMethodology <- function(jaspResults, dataset, options, ...){
   op2  <- length(options[["rsmResponseVariables"]])
   op3  <- length(options[["rsmBlocks"]])
 
-  if (options[["designType"]] == "cube" && options[["buildDesignInv"]])
-    .cubeDesign(jaspResults, options, dataset)
-  if (options[["designType"]] == "star" && options[["buildDesignInv"]])
-    .starDesign(jaspResults, options)
+  placeholder <- createJaspTable(title = gettext("Response Surface Methodology"))
+  jaspResults[["placeholder"]] <- placeholder
+  placeholder$dependOn(options = c("rsmVariables","rsmResponseVariables", "contour","coef","anova","res","pareto","resNorm",
+                                   "resFitted","buildDesignInv","desirability","contour"))
+  rsm <- list()
+  #Placeholder table when the user inputs something. If an analysis gets picked, remove the table
+  if((op1 == 0 || op2 == 0)) {
+
+    if (((op1 > 0 && op2 == 0) | (op1 == 0 && op2 > 0)) & any(options[["contour"]], options[["coef"]], options[["anova"]],
+                                                                options[["res"]], options[["pareto"]], options[["resNorm"]], options[["ResFitted"]],
+                                                                options[["desirability"]],
+                                                                options[["contour"]])) {
```

Comment:
These checks look incredibly random. Can't you make one variable called `ready` and hide all the logic in there?